File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int res = 0;
        int buyPrice = prices[0];
        for(int i=1;i<prices.size();i++){
            buyPrice = min(buyPrice, prices[i]);
            res = max(res, prices[i]-buyPrice);
        }
        
        return res;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        
        min_price = prices[0]
        price = 0

        for i in range (1, len(prices)):
            
            if (min_price > prices[i]):
                min_price = prices[i]

            price = max(price, prices[i] - min_price)


        return price